crates/rspack_core/src/compiler/compilation.rs

```diff
@@ -1394,8 +1394,19 @@ impl Compilation {
     (path, info)
   }
 
-  pub fn get_asset_path(&self, filename: &Filename, data: PathData) -> String {
-    filename.render(data, None)
+  pub fn get_asset_path(&self, filename: &Filename, data: PathData<'_>) -> String {
+    let rendered_filename = filename.render(data, None);
+    return tokio::task::block_in_place(move || {
+      tokio::runtime::Handle::current().block_on(async move {
+        self
+          .plugin_driver
+          .write()
```

Comment:
This will be stuck if you would like to call this method from a JS Hook. As `write` was called as Rust called the JS hook, and if you call this method, then, Rust calls the `write` of `plugin_driver` again, which leads to a deadlock. The only way to solve this is to call the `plugin_driver` as `read`.

Reply:
> Are there any other ways to overcome this issue? As I can see we can't really invoke the hook from elsewhere or at another point of time, right?

If you want to acquire the mutable reference as what `write` does, you can only invoke the hook one at the time. But for hooks that only requires `read`, it does not have the limitation, which means you can invoke as much as you wish.

Reply:
@h-a-n-a can you please point me what exactly should be changed in my implementation? I tried both read and write accesses and behavior is the same

Reply:
The deadlock problem has been resolved in https://github.com/web-infra-dev/rspack/pull/3455. You may continue to implement this hook. Thank you in advance!

Reply:
Thanks a lot @h-a-n-a !